app/scripts/directives/mobileServiceInstanceRow.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+(function() {
+  angular.module('openshiftConsole').component('mobileServiceInstanceRow', {
+    controller: [
+      '$filter',
+      '$routeParams',
+      '$scope',
+      'APIService',
+      'AuthorizationService',
+      'BindingService',
+      'Constants',
+      'DataService',
+      'ListRowUtils',
+      'ServiceInstancesService',
+      MobileServiceInstanceRow
+    ],
+    controllerAs: 'row',
+    bindings: {
+        apiObject: '<',
+        serviceClass: '<',
+        binding: '<',
+        state: '<'
+    },
+    templateUrl: 'views/_mobile-service-instance-row.html'
+  });
+
+  function MobileServiceInstanceRow($filter,
+                              $routeParams,
+                              $scope,
+                              APIService,
+                              AuthorizationService,
+                              BindingService,
+                              Constants,
+                              DataService,
+                              ListRowUtils,
+                              ServiceInstancesService) {
+    var row = this;
+    var serviceInstanceDisplayName = $filter('serviceInstanceDisplayName');
+    row.serviceBindingsVersion = APIService.getPreferredVersion('servicebindings');
+    row.state = row.state || {};
```

Comment:
Why is this needed?

Reply:
state is expected by the ListRowUtils.ui, and is not always set, this initiates it to an empty object if it is not passed in.